homeassistant/components/awair/__init__.py

```diff
@@ -1 +1,106 @@
+from homeassistant.core import Config, HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import API_TIMEOUT, DOMAIN, LOGGER, UPDATE_INTERVAL, AwairResult
+
+PLATFORMS = ["sensor"]
+
+
+async def async_setup(hass: HomeAssistant, config: Config) -> bool:
+    """Set up Awair integration."""
+    return True
+
+
+async def async_setup_entry(hass, config_entry) -> bool:
+    """Set up Awair integration from a config entry."""
+    access_token = config_entry.data[CONF_ACCESS_TOKEN]
+    session = async_get_clientsession(hass)
+    coordinator = AwairDataUpdateCoordinator(hass, access_token, session)
+
+    await coordinator.async_refresh()
+
+    if not coordinator.last_update_success:
+        raise ConfigEntryNotReady
+
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][config_entry.entry_id] = coordinator
+
+    for platform in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(config_entry, platform)
+        )
+
+    return True
+
+
+async def async_unload_entry(hass, config_entry) -> bool:
+    """Unload Awair configuration."""
+    tasks = []
+    for platform in PLATFORMS:
+        tasks.append(
+            hass.config_entries.async_forward_entry_unload(config_entry, platform)
+        )
+
+    unload_ok = all(await gather(*tasks))
+    if unload_ok:
+        hass.data[DOMAIN].pop(config_entry.entry_id)
+
+    return unload_ok
+
+
+class AwairDataUpdateCoordinator(DataUpdateCoordinator):
+    """Define a wrapper class to update Awair data."""
+
+    def __init__(self, hass, access_token, session) -> None:
+        """Set up the AwairDataUpdateCoordinator class."""
+        self.__awair = Awair(access_token=access_token, session=session)
+
+        super().__init__(hass, LOGGER, name=DOMAIN, update_interval=UPDATE_INTERVAL)
+
+    async def _async_update_data(self) -> Optional[Any]:
+        """Update data via Awair client library."""
+        with timeout(API_TIMEOUT):
+            try:
+                LOGGER.debug("Fetching users and devices")
+                user = await self.__awair.user()
+                devices = await user.devices()
+                results = await gather(
+                    *[self.__fetch_air_data(device) for device in devices]
+                )
+                return {result.device.uuid: result for result in results}
+            except RatelimitError as err:
+                raise UpdateFailed(err)
+            except AwairError as err:
+                message = (
+                    "Unable to update Awair data - please ensure your access token is correct. ",
+                    "You may unload and re-add the Awair integration if you need to update the access token.",
+                )
+                persistent_notification.async_create(
```

Comment:
We should start a reauth step in the config flow instead. See eg:
https://github.com/home-assistant/core/blob/b3c851502d3e5f119632429edc4abb00650d1251/homeassistant/components/samsungtv/media_player.py#L110-L112